src/main/java/org/terasology/launcher/gui/javafx/SettingsController.java

```diff
@@ -433,15 +434,23 @@ private void populateLanguageValues() {
         languageBox.getItems().clear();
         for (Locale locale : Languages.SUPPORTED_LOCALES) {
             String item = locale.toLanguageTag() + " : " + BundleUtils.getLabel(locale, Languages.SETTINGS_LABEL_KEYS.get(locale));
-            if (!locale.equals(Languages.getCurrentLocale())) {
-                item += " (" + BundleUtils.getLabel(Languages.SETTINGS_LABEL_KEYS.get(locale)) + ")";
-            }
-            languageBox.getItems().add(item);
 
-            if (Languages.getCurrentLocale().equals(locale)) {
-                languageBox.getSelectionModel().select(item);
-            }
+
+                if (!locale.equals(Languages.getCurrentLocale())) {
+                    item += " (" + BundleUtils.getLabel(Languages.SETTINGS_LABEL_KEYS.get(locale)) + ")";
+                }
+                Collator coll = Collator.getInstance(locale);
+                languageBox.getItems().add(item);
+                languageBox.getItems().sort(coll);
```

Comment:
The list doesn't have to be sorted whenever an item is added - one sort() call after it's fully populated will work just as well.